daffodil-core/src/main/scala/org/apache/daffodil/dsom/RawCommonRuntimeValuedPropertiesMixin.scala

```diff
@@ -49,69 +49,68 @@ trait RawElementRuntimeValuedPropertiesMixin
+  protected final lazy val optionExtraEscapedCharactersRaw = findPropertyOption("extraEscapedCharacters", expressionAllowed = true)
   protected final lazy val extraEscapedCharactersRaw = requireProperty(optionExtraEscapedCharactersRaw)
 
 }
 
 trait RawSimpleTypeRuntimeValuedPropertiesMixin
   extends RawCommonRuntimeValuedPropertiesMixin {
 
-  protected final lazy val optionTextStandardDecimalSeparatorRaw = findPropertyOption("textStandardDecimalSeparator")
+  protected final lazy val optionTextStandardDecimalSeparatorRaw = findPropertyOption("textStandardDecimalSeparator", expressionAllowed = true)
   protected final lazy val textStandardDecimalSeparatorRaw = requireProperty(optionTextStandardDecimalSeparatorRaw)
-  protected final lazy val optionTextStandardGroupingSeparatorRaw = findPropertyOption("textStandardGroupingSeparator")
+  protected final lazy val optionTextStandardGroupingSeparatorRaw = findPropertyOption("textStandardGroupingSeparator", expressionAllowed = true)
   protected final lazy val textStandardGroupingSeparatorRaw = requireProperty(optionTextStandardGroupingSeparatorRaw)
 
   
-  protected final lazy val optionBinaryFloatRepRaw = findPropertyOption("binaryFloatRep")
+  protected final lazy val optionBinaryFloatRepRaw = findPropertyOption("binaryFloatRep", expressionAllowed = true)
   protected final lazy val binaryFloatRepRaw = requireProperty(optionBinaryFloatRepRaw)
-  protected final lazy val optionTextBooleanTrueRepRaw = findPropertyOption("textBooleanTrueRep")
+  protected final lazy val optionTextBooleanTrueRepRaw = findPropertyOption("textBooleanTrueRep", expressionAllowed = true)
   protected final lazy val textBooleanTrueRepRaw = requireProperty(optionTextBooleanTrueRepRaw)
-  protected final lazy val optionTextBooleanFalseRepRaw = findPropertyOption("textBooleanFalseRep")
+  protected final lazy val optionTextBooleanFalseRepRaw = findPropertyOption("textBooleanFalseRep", expressionAllowed = true)
   protected final lazy val textBooleanFalseRepRaw = requireProperty(optionTextBooleanFalseRepRaw)
-  protected final lazy val optionCalendarLanguageRaw = findPropertyOption("calendarLanguage")
+  protected final lazy val optionCalendarLanguageRaw = findPropertyOption("calendarLanguage", expressionAllowed = true)
   protected final lazy val calendarLanguageRaw = requireProperty(optionCalendarLanguageRaw)
-  protected final lazy val optionBinaryBooleanTrueRepRaw = findPropertyOption("binaryBooleanTrueRep")
+  protected final lazy val optionBinaryBooleanTrueRepRaw = findPropertyOption("binaryBooleanTrueRep", expressionAllowed = true)
   protected final lazy val binaryBooleanTrueRepRaw = requireProperty(optionBinaryBooleanTrueRepRaw)
-  protected final lazy val optionBinaryBooleanFalseRepRaw = findPropertyOption("binaryBooleanFalseRep")
+  protected final lazy val optionBinaryBooleanFalseRepRaw = findPropertyOption("binaryBooleanFalseRep", expressionAllowed = true)
```

Comment:
binaryBooleanTrueRep and binaryBooleanFalseRep cannot be an expression:

https://daffodil.apache.org/docs/dfdl/#_Toc398030762

Reply:
I've updated the commit to remove the allowedExpression=true from the lines where it was incorrect.